### PR TITLE
Move exampels to separate crate

### PR DESCRIPTION
**Pull request by haerdib**
This way, the examples don't need to inherit all the features of the api-client.  They import the api-client with the necessary features. This serves as a better example on how to import and use the api-client.

### CODE REVIEW

**Review by echevrier — COMMENTED** [review]
I don't like the repetition examples/examples. Instead we could regroup the examples:
examples/
- compose_extrinsic/
- events/
- storage/
...

**Comment by haerdib** [comment]
That sadly doesn't work because of cargo. To name the second example folder `examples` is necessary to mark them as `examples`. Otherwise, each examples of them would need to be defined in the .toml file as `[[example]]` (see https://doc.rust-lang.org/cargo/reference/cargo-targets.html#examples)

**Review by echevrier — APPROVED** [review]
Then :)

File: examples/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+frame-support = { git = "https://github.com/paritytech/substrate.git", branch = "master" }
+frame-system = { git = "https://github.com/paritytech/substrate.git", branch = "master" }
+kitchensink-runtime = { git = "https://github.com/paritytech/substrate.git", branch = "master" }
+pallet-balances = { git = "https://github.com/paritytech/substrate.git", branch = "master" }
+pallet-identity = { git = "https://github.com/paritytech/substrate.git", branch = "master" }
+pallet-staking = { git = "https://github.com/paritytech/substrate.git", branch = "master" }
+sp-core = { features = ["full_crypto"], git = "https://github.com/paritytech/substrate.git", branch = "master" }
+sp-keyring = { git = "https://github.com/paritytech/substrate.git", branch = "master" }
+sp-runtime = { git = "https://github.com/paritytech/substrate.git", branch = "master" }
+sp-version = { git = "https://github.com/paritytech/substrate.git", branch = "master" }
```
Comment (clangenb) [inline examples/Cargo.toml:25]: I like that these deps no longer live in the root workspace.

**Review by clangenb — APPROVED** [review]
Nice, lgtm!

**Pull request merged**